chain/network/src/peer/peer_actor.rs

```diff
@@ -241,67 +170,42 @@ impl PeerActor {
         // Start PeerActor on separate thread.
-        Ok(Self::start_in_arbiter(&actix::Arbiter::new().handle(), move |ctx| {
-            let pa = Self {
-                clock,
-                my_node_info,
-                peer_addr,
-                peer_type: match &stream_config {
-                    StreamConfig::Inbound => PeerType::Inbound,
-                    StreamConfig::Outbound { .. } => PeerType::Outbound,
-                },
-                peer_status: PeerStatus::Connecting(connecting_status),
-                protocol_version: PROTOCOL_VERSION,
-                framed: FramedWrite::new(
-                    write,
-                    Codec::new(write_buf_size.clone()),
-                    Codec::new(write_buf_size),
-                    ctx,
-                ),
-                tracker: Default::default(),
-                stats: Arc::new(connection::Stats::default()),
-                partial_edge_info: match &stream_config {
-                    StreamConfig::Inbound => None,
-                    StreamConfig::Outbound { peer_id } => {
-                        Some(network_state.propose_edge(peer_id, None))
-                    }
-                },
-                routed_message_cache: LruCache::new(ROUTED_MESSAGE_CACHE_SIZE),
-                protocol_buffers_supported: false,
-                force_encoding,
-                peer_info: match &stream_config {
-                    StreamConfig::Inbound => None,
-                    StreamConfig::Outbound { peer_id } => Some(PeerInfo {
-                        id: peer_id.clone(),
-                        addr: Some(peer_addr.clone()),
-                        account_id: None,
-                    }),
-                }
-                .into(),
-                network_state,
-                connection: None,
-                _connection_guard: connection_guard,
-            };
-            let addr = ctx.address();
-            let stats = pa.stats.clone();
-            ctx.spawn(wrap_future(async move {
-                if let Err(err) =
-                    run_recv_loop(read, peer_addr, addr.clone().recipient(), stats).await
-                {
-                    // Send the critical error to the PeerActor (best effort).
-                    // PeerActor is responsible for calling ctx.stop() afterwards.
-                    addr.do_send(RecvMessage(Err(err)));
+        Ok(Self::start_in_arbiter(&actix::Arbiter::new().handle(), move |ctx| Self {
+            clock,
+            my_node_info,
+            peer_addr,
+            peer_type: match &stream_config {
+                StreamConfig::Inbound => PeerType::Inbound,
+                StreamConfig::Outbound { .. } => PeerType::Outbound,
+            },
+            peer_status: PeerStatus::Connecting(connecting_status),
+            protocol_version: PROTOCOL_VERSION,
+            framed: stream::FramedStream::spawn(ctx, peer_addr, stream, stats.clone()),
```

Comment:
I'd maybe `let framed =` this -- side-effectful op in a record literall feels a bit unclean. Super weak opinion though.

Reply:
done